common/src/main/java/com/microsoft/hydralab/common/management/AppiumServerManager.java

```diff
@@ -111,7 +113,16 @@ public IOSDriver getIOSDriver(DeviceInfo deviceInfo, Logger logger) {
         if (iosDriver != null && isDriverAlive(iosDriver)) {
             logger.info(iosDriver.toString());
             logger.info(iosDriver.getStatus().toString());
-            return iosDriver;
+            if (isDriverExpired(deviceInfo)) {
+                try {
+                    logger.info("driver expired, quit old driver and create a new one");
+                    quitIOSDriver(deviceInfo, logger);

```

Comment:
Does the iOS driver automatically reconnect after being quit?

Reply:
The original session would be killed and regenerate a new one.